virtcontainers/kata_agent.go

```diff
@@ -777,16 +776,43 @@ func (k *kataAgent) createContainer(sandbox *Sandbox, c *Container) (p *Process,
 	// We need to give the OCI spec our absolute rootfs path in the guest.
 	grpcSpec.Root.Path = rootPath
 
+	sharedPidNs := false
+	pidNsIndex := -1
+
+	// Sandbox pid will be set once the sandbox container is created.
+	// Check if pid namespace path is the same as the sandbox/infra container
+	if sandbox.state.Pid > 0 {
+		for i, ns := range grpcSpec.Linux.Namespaces {
```

Comment:
Ok I understand that given the current architecture, this is the only thing we can do. But I think we should handle the detection at the `container.go` level, meaning more generically so that it can be reused across agent implementation and provided directly through a new boolean field of the `Container` structure.
Now, this is currently not feasible because this means we would introduce some OCI assumptions into `container.go`. The way to avoid that is to parse the namespaces provided by the OCI spec from `virtcontainers/pkg/oci/utils.go`, and fill the `ContainerConfig` with those information.
This way, we could analyze the namespaces expected for the container and determine if we have to share the PID namespace, or any other namespace from `container.go`.
@amshinde WDYT ?

Reply:
@sboeuf Sounds reasonable to check for shared namespaces in a generic way for reuse by the agents. I'll rework this.